Add Cookstyle 5.13 notes (#414)
chef/cookstyle: RELEASE_NOTES.md

## Cookstyle 5.13

### 3 New Chef Cops

#### ChefDeprecations/LegacyNotifySyntax

> The `ChefDeprecations/LegacyNotifySyntax` cop detects the legacy notification syntax in resources. The newer notification suppports notifying resources that occur later in the recipe and should be used instead.

Reviewer: ```suggestion
The `ChefDeprecations/LegacyNotifySyntax` cop detects the legacy notification syntax in resources. The newer notification method supports notifying resources that occur later in the recipe and should be used instead.
```

> Legacy notification syntax

Reviewer: ```suggestion
Legacy notification syntax:
```

```ruby
notifies :restart, resources(service: 'apache')
```

> New notification syntax

Reviewer: ```suggestion
New notification syntax:
```

```ruby
notifies :restart, 'service[apache]'
```

`Enabled by default`: True

`Autocorrects`: Yes

#### ChefDeprecations/NodeSetWithoutLevel

The `ChefDeprecations/NodeSetWithoutLevel` cop detects recipes and resources that set node attributes without specifying the precedence level required by Chef Infra Client 11 and later.

> Setting node attributes without a precedence level

Reviewer: ```suggestion
Setting node attributes without a precedence level:
```

```ruby
node['foo']['bar'] = 1
node['foo']['bar'] << 1
node['foo']['bar'] += 1
node['foo']['bar'] -= 1
```

> Setting node attributes with a precedence level of override

Reviewer: ```suggestion
Setting node attributes with a precedence level of override:
```

> ```ruby

Reviewer: I think there's supposed to be an empty line before a codeblock. 

```suggestion

```ruby
```
